### PR TITLE
feat: support hook config objects in warp config

### PR DESCRIPTION
**Pull request by yorhodes**
### Description

- Adds support for hook config objects on warp config
- Adds core config schema

### Drive-by changes

- Removes ICA config from schemas (resolve at config time)

### Related issues

- Fixes https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3728
- Fixes https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3667

### CODE REVIEW

File: typescript/sdk/src/core/schemas.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { HookConfigSchema } from '../hook/schemas.js';
+import { IsmConfigSchema } from '../ism/schemas.js';
+import { OwnableSchema } from '../schemas.js';
+
+export const CoreConfigSchema = OwnableSchema.extend({
```
Comment (jmrossy) [inline typescript/sdk/src/core/schemas.ts:5]: 👍

File: typescript/sdk/src/deploy/HyperlaneAppChecker.ts [hunk 2]
```diff
@@ -208,19 +206,12 @@ export abstract class HyperlaneAppChecker<
...
     chain: ChainName,
-    owner: Owner,
+    owner: Address,
     ownableOverrides?: Record<string, Address>,
   ): Promise<void> {
     const ownableContracts = await this.ownables(chain);
     for (const [name, contract] of Object.entries(ownableContracts)) {
       let expectedOwner = ownableOverrides?.[name] ?? owner;
-      if (typeof expectedOwner !== 'string') {
-        expectedOwner = await resolveOrDeployAccountOwner(
```
Comment (jmrossy) [inline typescript/sdk/src/deploy/HyperlaneAppChecker.ts:218]: Can you confirm we don't want to check this?

File: typescript/sdk/src/gas/oracle/types.ts [hunk 3]
```diff
@@ -43,9 +47,14 @@ const serializePercentDifference = (
   return diff.isNegative() ? `${diff.toString()}%` : `+${diff.toString()}%`;
 };
 
+export const bignumberify = (config: StorageGasOracleConfig): OracleData => ({
```
Comment (jmrossy) [inline typescript/sdk/src/gas/oracle/types.ts:50]: Please rename this. `oracleConfigToOracleData`?

File: typescript/sdk/src/deploy/HyperlaneDeployer.ts [hunk 4]
```diff
@@ -291,32 +290,31 @@ export abstract class HyperlaneDeployer<
   protected async configureHook<C extends Ownable>(
     chain: ChainName,
     contract: C,
-    targetHook: IPostDispatchHook,
+    config: HookConfig,
     getHook: (contract: C) => Promise<Address>,
     setHook: (contract: C, hook: Address) => Promise<PopulatedTransaction>,
   ): Promise<void> {
+    if (typeof config !== 'string') {
```
Comment (jmrossy) [inline typescript/sdk/src/deploy/HyperlaneDeployer.ts:297]: 😢

File: typescript/sdk/src/hook/EvmHookReader.ts [hunk 5]
```diff
@@ -118,8 +117,8 @@ export class EvmHookReader implements HookReader {
   async deriveMerkleTreeConfig(
     address: Address,
   ): Promise<WithAddress<MerkleTreeHookConfig>> {
-    const hook = MerkleTreeHook__factory.connect(address, this.provider);
-    assert((await hook.hookType()) === OnchainHookType.MERKLE_TREE);
+    // const hook = MerkleTreeHook__factory.connect(address, this.provider);
+    // assert((await hook.hookType()) === OnchainHookType.MERKLE_TREE);
```
Comment (jmrossy) [inline typescript/sdk/src/hook/EvmHookReader.ts:121]: Why remove these? WIP?

**Review by jmrossy — COMMENTED** [review]
The changes mostly lgtm and it's nice to have more red than green here. 
And don't forget to add a changeset

File: typescript/sdk/src/hook/EvmHookReader.ts (hunk 5 above)
Comment (yorhodes) [reply typescript/sdk/src/hook/EvmHookReader.ts:121]: will revert these

File: typescript/sdk/src/deploy/HyperlaneDeployer.ts (hunk 4 above)
Comment (yorhodes) [reply typescript/sdk/src/deploy/HyperlaneDeployer.ts:297]: punting this until https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/3773

File: typescript/sdk/src/deploy/HyperlaneAppChecker.ts (hunk 2 above)
Comment (yorhodes) [reply typescript/sdk/src/deploy/HyperlaneAppChecker.ts:218]: yes, I removed this from the type

**Comment by yorhodes** [comment]
> And don't forget to add a changeset. 

Will do

> Also can you confirm this shouldn't be to the `cli-2.0` branch?

Good point, will rebase

**Review by jmrossy — APPROVED** [review]

**Pull request merged**